internal/controller/atlas/provider.go

```diff
@@ -43,24 +38,28 @@ type ClientSet struct {
 }
 
 type ProductionProvider struct {
-	k8sClient       client.Client
-	domain          string
-	globalSecretRef client.ObjectKey
-	dryRun          bool
+	domain string
+	dryRun bool
 }
 
-type credentialsSecret struct {
+// Credentials is the type that holds credentials to authenticate against the Atlas API.
+// Currently, only API keys are support but more credential types could be added,
+// see https://www.mongodb.com/docs/atlas/configure-api-access/.
+type Credentials struct {
```

Comment:
I'd have one struct (merge `Credentials` and `ApiKeys`) and remove `OrgID` field.
The org id is not needed to setup the client and the function would be only giving it back to the caller.

Reply:
No, i disagree, we have to be open to add additional key types in the future. This struct prepares us to be ready for that.

Reply:
but, I think you have one good point though, I believe we can remove the `orgID` at all from the credentials setup 🤔  Let me check if that is possible 👍